firebase_push_notifications/firebase_push_notifications/v1_0/manager.py

```diff
@@ -0,0 +1,148 @@
+"""This module contains the logic for sending push notifications to the user's device using the Firebase Cloud Messaging service."""  # noqa: E501
+import json
+import logging
+import os
+from datetime import timedelta
+
+import google.auth.transport.requests
+import requests
+from aries_cloudagent.messaging.util import datetime_now, time_now
+from aries_cloudagent.storage.base import StorageNotFoundError
+from dateutil import parser
+from google.oauth2 import service_account
+
+from .constants import (
+    BASE_URL,
+    ENDPOINT_PREFIX,
+    ENDPOINT_SUFFIX,
+    MAX_SEND_RATE_MINUTES,
+    SCOPES,
+)
+from .models import FirebaseConnectionRecord
+
+PROJECT_ID = os.environ.get("FIREBASE_PROJECT_ID", "")
+FCM_ENDPOINT = ENDPOINT_PREFIX + PROJECT_ID + ENDPOINT_SUFFIX
+FCM_URL = BASE_URL + "/" + FCM_ENDPOINT
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_access_token():
+    """Gets an access token for sending firebase messages with the service account credentials."""  # noqa: E501
```

Comment:
Small nitpick but usually the linter complains if there's no space between the docstring and the first line of code. I'm guessing that if you mark the line with `# noqa` the linter doesn't complain